x/stake/keeper/delegation.go

```diff
@@ -664,21 +673,42 @@ func (k Keeper) BeginUnbonding(ctx sdk.Context,
 
 // complete unbonding an unbonding record
 // CONTRACT: Expects unbonding passed in has finished the unbonding period
-func (k Keeper) CompleteUnbonding(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress) (types.UnbondingDelegation, sdk.Error) {
+func (k Keeper) CompleteUnbonding(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress) (types.UnbondingDelegation, sdk.Events, sdk.Error) {
 	ubd, found := k.GetUnbondingDelegation(ctx, delAddr, valAddr)
 	if !found {
-		return ubd, types.ErrNoUnbondingDelegation(k.Codespace())
+		return ubd, sdk.Events{}, types.ErrNoUnbondingDelegation(k.Codespace())
 	}
 
-	_, err := k.bankKeeper.SendCoins(ctx, DelegationAccAddr, ubd.DelegatorAddr, sdk.Coins{ubd.Balance})
-	if err != nil {
-		return ubd, err
+	var events sdk.Events
+	if ubd.CrossStake {
+		_, err := k.BankKeeper.SendCoins(ctx, DelegationAccAddr, ubd.DelegatorAddr, sdk.Coins{ubd.Balance})
+		if err != nil {
+			return ubd, sdk.Events{}, err
+		}
+		bondDenom := k.BondDenom(ctx)
+		balance := k.BankKeeper.GetCoins(ctx, ubd.DelegatorAddr).AmountOf(bondDenom)
+		_, err = k.BankKeeper.SendCoins(ctx, ubd.DelegatorAddr, sdk.PegAccount, sdk.Coins{sdk.NewCoin(bondDenom, balance)})
```

Comment:
Please put `SendCoins` inside `crossDistributeUndelegated`, the flow of funds maybe more resonable.

Reply:
Done